### ISSUE TITLE
Component set to required does not have `required` or `aria-required` label on the focused element

### ISSUE THREAD

**Issue by fendiaoxiaoshuazi** [Esri/calcite-components#13271]
**monday.com sync:** #10513353201

### Check existing issues

- [x] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates and reviewed the [Accessibility page](https://developers.arcgis.com/calcite-design-system/foundations/accessibility) for guidance.

### Which Component

Input Date Picker, Input Time Picker, Slider

### Actual Behavior

Component set to required does not have `required` or `aria-required` label on the focused element 
<img width="2555" height="1359" alt="Image" src="https://github.com/user-attachments/assets/1b2033e4-10eb-477b-bd25-ae95749f06ac" />
This issue was raised with the Survey123 team
<img width="726" height="260" alt="Image" src="https://github.com/user-attachments/assets/23142aba-4911-4ab0-af22-73e163f4aac9" />

### Expected Behavior

Add `aria-required` label so that the screen reader applications can announce the info

### Reproduction Sample

https://developers.arcgis.com/calcite-design-system/components/slider/#sample

### Reproduction Steps

1. Check the required option
2. Use NVDA and navigate to the component
3. Observe that "required" is not announcted

### Reproduction Version

3.3.3

### Working W3C Example/Tutorial

https://www.w3.org/WAI/WCAG22/Techniques/aria/ARIA2

### Relevant Info

NVDA cannot identify the required field.

### Priority impact

impact - p2 - want for an upcoming milestone

### Calcite package

- [X] @esri/calcite-components
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

ArcGIS Survey123

**Comment by fendiaoxiaoshuazi** [comment]
There are 2 support issues related to this feature request:
- BUG-000179415 
<img width="1132" height="527" alt="Image" src="https://github.com/user-attachments/assets/6fd65f33-6a0d-46fb-b70a-c586ab181086" />

- BUG-000179007

<img width="1422" height="519" alt="Image" src="https://github.com/user-attachments/assets/2fa183de-9d01-4210-aadb-f6a4df8c9198" />
